documentation/create_overview_doc/createdoc.sh

```diff
@@ -111,419 +82,215 @@ function read_skip_list() {
-    if [[ -n "$mermaid_diagram" && "$mermaid_diagram" =~ [^[:space:]] ]]; then 
-        echo -e "## Flow Diagram [via mermaid]\n\`\`\`mermaid\n$mermaid_diagram\n\`\`\`" >> "$markdown_documentation_file"
-    fi 
 
-    echo -e "Documentation saved to $markdown_documentation_file\n\n"
+    # Generate the flow chart for the module
+    generate_flow_chart "$path_to_module" "$documentation_directory/flowmaps"
+
+    # Add link to the flow chart in the documentation
+    local flow_chart_name=$(basename "$path_to_module" | cut -f 1 -d '.')
+    echo -e "\n![Flow Chart of $flow_chart_name](./flowmaps/${flow_chart_name}_FlowChart.png)\n" >> "$markdown_documentation_file"
+
+    echo -e "Documentation saved to $markdown_documentation_file\n\n" >&2
 }
 
-function extract_module_names_and_associated_objectives_then_call_bito() {
-    local filename=$1
-    local prompt_file_path=$2
-    local lines=()
-    local current_module=""
-    local current_objectives=""
-    local capture_objectives=false
-    local combined_output=""
-    local attempt=1
-    local MAX_RETRIES=5
-    local RETRY_DELAY=10 # seconds
-    local bito_output
-
-    echo -e "Extracting module names and objectives from file: $filename\n" >&2
-
-    # Read the file line by line into the array
-    while IFS= read -r line; do
-        lines+=("$line")
-    done < "$filename"
-
-    # Loop through the lines to process them
+function extract_module_details_and_call_bito() {
+    local filename=$1 prompt_file_path=$2 lines=() current_module="" current_objectives="" current_operational_sequence="" capture_objectives=false capture_operational_sequence=false combined_output=""
+
+    echo "Compiling objectives and details for module in $filename..." >&2
+
+    while IFS= read -r line; do lines+=("$line"); done < "$filename"
+
     for line in "${lines[@]}"; do
         if [[ $line =~ ^##\ Module:\ (.*) ]]; then
             if [[ -n $current_module ]]; then
-                echo "Processing module: $current_module with objectives" >&2
-                combined_output+="Module: $current_module\n---\nPrimary Objectives:\n$current_objectives\n\n"
+                echo "Processing module: $current_module with objectives and operational sequence" >&2
+                combined_output+="Module: $current_module\n---\nPrimary Objectives:\n$current_objectives\nOperational Sequence:\n$current_operational_sequence\n\n"
             fi
             current_module=${BASH_REMATCH[1]}
-            current_objectives=""
-            capture_objectives=false
+            current_objectives="" current_operational_sequence=""
+            capture_objectives=false capture_operational_sequence=false
         elif [[ $line =~ ^-\ \*\*Primary\ Objectives\*\*: ]]; then
             capture_objectives=true
+            capture_operational_sequence=false
             current_objectives+=$(echo $line | sed 's/.*\*\*Primary Objectives\*\*: //')$'\n'
+        elif [[ $line =~ ^-\ \*\*Operational\ Sequence\*\*: ]]; then
```

Comment:
**Scalability Issue**: The retry logic for calling external commands is a critical component for scalability, ensuring resilience against transient failures. However, the hard-coded retry limits and delay may not be optimal for all environments or situations. <br> **Fix**: Externalize configuration of retry parameters (MAX_RETRIES, RETRY_DELAY) to allow flexible adjustment based on environment or runtime conditions. <br> **Code Suggestion**: 
 ```
 No Code Suggestion
 ```